lib/nx_deflate.c

```diff
@@ -1655,14 +1643,11 @@ int nx_deflate(z_streamp strm, int flush)
 		if (!(s->status & (NX_BFINAL_ST | NX_TRAILER_ST))) {
 			if (s->avail_out == 0)
 				return TRACERET(Z_OK); /* need more output space */
-			if ((s->used_out == 0) && (s->avail_in == 0))
-				return TRACERET(Z_OK); /* no input here */
 		}
-	}
-
-	/* check if stream end has been reached */
-	if (s->avail_in == 0 && s->used_in == 0 && s->used_out == 0)
-		return nx_stream_end(s);
+	/* TODO: zlib also compares previous and current flush to avoid
+	 * duplicate consecutive flushes. */
+	} else if (s->avail_in == 0 && (flush < Z_PARTIAL_FLUSH || flush > Z_FINISH))
```

Comment:
A comment explaining (flush < Z_PARTIAL_FLUSH || flush > Z_FINISH) would be good.

Reply:
@rzinsly what about now?

This is only a faster way to check `flush != Z_PARTIAL_FLUSH && flush != Z_SYNC_FLUSH && flush != Z_FULL_FLUSH && flush != Z_FINISH`

Reply:
Yes, but was not obvious, LGTM now.